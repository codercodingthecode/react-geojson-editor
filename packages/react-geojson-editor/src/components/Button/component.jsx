import React from 'react'; // eslint-disable-line import/no-extraneous-dependencies
import PropTypes from 'prop-types'; // eslint-disable-line import/no-extraneous-dependencies

import ButtonStyles from './styles.scss';

export const Button = ({ children, ...props }) => (
  <button
    className={ButtonStyles.button}
    {...props}
  >
    {children}
  </button>
);

Button.propTypes = {
  children: PropTypes.string,
};

Button.defaultProps = {
  children: 'Button',
};

export default Button;
