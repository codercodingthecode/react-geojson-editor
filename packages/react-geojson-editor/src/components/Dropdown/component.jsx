/* eslint-disable */

import React, { Component, Children } from 'react'; // eslint-disable-line import/no-extraneous-dependencies
import PropTypes from 'prop-types'; // eslint-disable-line import/no-extraneous-dependencies
import classnames from 'classnames'; // eslint-disable-line import/no-extraneous-dependencies

// import OutsideClickHandler from 'react-outside-click-handler';
import DropdownStyles from './styles.scss';

export class Dropdown extends Component {
    state = {
      open: this.props.openAtStart,
    }
    constructor(props) {
      super(props);
      this.onBlurClose = this.onBlurClose.bind(this);
      this.toggleOpen = this.toggleOpen.bind(this);
      this.renderContent = this.renderContent.bind(this);
    }

    onBlurClose() {
      this.setState({ open: false });
    }

    toggleOpen(e) {
      e.preventDefault();
      const { children } = this.props;
      this.setState({ open: !this.state.open && Boolean(Children.count(children)) });
    }

    renderContent(children) {
      return (
        <div className={DropdownStyles.dropdownBody}>
          {children}
        </div>
      );
    }

    render() {
      const { open } = this.state;
      const {
        children, name, positionRight, style, containerStyle,
      } = this.props;
      return (
        <div
          className={
                positionRight ?
                    classnames(DropdownStyles.mlDropdown, DropdownStyles.rightPosition) :
                    DropdownStyles.mlDropdown
            }
          style={style}
        >
          <div className={DropdownStyles.dropdownContainer} style={containerStyle}>
            {/* <OutsideClickHandler onOutsideClick={this.onBlurClose}> */}
            <div
              className={open ?
                                classnames(DropdownStyles.dropdownButton, DropdownStyles.open, 'toggle-button') :
                                classnames(DropdownStyles.dropdownButton, 'toggle-button')
                            }
              tabIndex="0"
              onClick={this.toggleOpen}
            >
              {name}
            </div>
            {open && this.renderContent(children)}
            {/* </OutsideClickHandler> */}
          </div>
        </div>
      );
    }
}

Dropdown.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.func,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
  name: PropTypes.string,
  style: PropTypes.object,
  containerStyle: PropTypes.object,
};

Dropdown.defaultProps = {
  name: 'Dropdown',
  style: {},
  containerStyle: {},
};

export default Dropdown;
