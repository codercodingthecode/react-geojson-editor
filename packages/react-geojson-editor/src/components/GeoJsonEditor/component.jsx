import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { GoogleMapWithLoader, PolygonDataProvider } from './google-map';
// import Reader from './geojson-manager/Reader';
import { GeoJsonEditorMode } from '../../utils';
import { GeoJsonType } from '../../custom-types';
import { PolygonController, GeoJsonController } from './controllers';
import { GeoJsonWriter } from './writer';
import { GeoJsonReader } from './reader';

export class GeoJsonEditor extends Component {
    state = {
      controlMode: GeoJsonEditorMode.VIEW,
    };

    render() {
      const {
        center, zoom, onSave, onDelete, existingPolygons, mapHeight, mapWidth,
      } = this.props;
      const { controlMode } = this.state;
      return (
        <div style={{ position: 'relative' }}>
          <GoogleMapWithLoader
            googleMapURL={`https://maps.googleapis.com/maps/api/js?key=${this.props.googleMapKey}&v=3.exp&libraries=geometry,drawing,places`}
            center={center}
            zoom={zoom}
            height={mapHeight}
            width={mapWidth}
          >
            <PolygonDataProvider
              initialGeojson={existingPolygons}
              onSave={onSave}
              onDelete={onDelete}
            >

              <GeoJsonController
                mode={controlMode}
                onToggleMode={mode => this.setState({ controlMode: mode })}
              />
              {controlMode === GeoJsonEditorMode.EDIT && <PolygonController />}

              {controlMode === GeoJsonEditorMode.EDIT ? <GeoJsonWriter /> : <GeoJsonReader />}

            </PolygonDataProvider>
          </GoogleMapWithLoader>
        </div>
      );
    }
}

export default GeoJsonEditor;

GeoJsonEditor.propTypes = {
  center: PropTypes.shape({
    lng: PropTypes.number,
    lat: PropTypes.number,
  }),
  zoom: PropTypes.number,
  googleMapKey: PropTypes.string,
  existingPolygons: GeoJsonType,
  onDelete: PropTypes.func,
  onSave: PropTypes.func,
  //   onUpdate: PropTypes.func,
  mapHeight: PropTypes.number,
  mapWidth: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
};
GeoJsonEditor.defaultProps = {
  center: { lng: -73.945, lat: 40.674 },
  zoom: 5,
  existingPolygons: null,
  googleMapKey: process.env.GOOGLE_MAP_KEY,
  mapHeight: 500,
  mapWidth: '100%',
  onDelete: () => {},
  onSave: () => {},
};
