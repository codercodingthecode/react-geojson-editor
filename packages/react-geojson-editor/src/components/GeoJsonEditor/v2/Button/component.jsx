import React, { Component, Children } from 'react';

import ButtonStyles from './styles.scss';

export const Button = ({ children, ...props }) => (
  <button
    className={ButtonStyles.button}
    {...props}
  >{children}
  </button>
);

export default Button;
