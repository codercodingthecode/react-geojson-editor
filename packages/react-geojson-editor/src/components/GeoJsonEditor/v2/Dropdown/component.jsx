import React, { Component, Children } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

// import OutsideClickHandler from 'react-outside-click-handler';
import DropdownStyles from './styles.scss';
import { StylesType } from '../../../../custom-types';

export function renderContent(children) {
  return (
    <div className={DropdownStyles.dropdownBody}>
      {children}
    </div>
  );
}

export class Dropdown extends Component {
  constructor(props) {
    super(props);
    this.onBlurClose = this.onBlurClose.bind(this);
    this.toggleOpen = this.toggleOpen.bind(this);
    // this.renderContent = this.renderContent.bind(this);
  }
    state = {
      open: this.props.openAtStart,
    }

    onBlurClose() {
      this.setState({ open: false });
    }

    toggleOpen(e) {
      e.preventDefault();
      const { children } = this.props;
      this.setState({ open: !this.state.open && Boolean(Children.count(children)) });
    }

    render() {
      const { open } = this.state;
      const {
        children, name, positionRight, style, containerStyle,
      } = this.props;
      return (
        <div
          className={
                positionRight ?
                    classnames(DropdownStyles.mlDropdown, DropdownStyles.rightPosition) :
                    DropdownStyles.mlDropdown
            }
          style={style}
        >
          <div className={DropdownStyles.dropdownContainer} style={containerStyle}>
            {/* <OutsideClickHandler onOutsideClick={this.onBlurClose}> */}
            <button
              className={
                open ?
                  classnames(DropdownStyles.dropdownButton, DropdownStyles.open) :
                  DropdownStyles.dropdownButton
              }
              onClick={this.toggleOpen}
            >
              {name}
            </button>
            {open && renderContent(children)}
            {/* </OutsideClickHandler> */}
          </div>
        </div>
      );
    }
}

Dropdown.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.func,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
  name: PropTypes.string,
  style: StylesType,
  containerStyle: StylesType,
  positionRight: PropTypes.bool,
  openAtStart: PropTypes.bool,
};

Dropdown.defaultProps = {
  name: 'Dropdown',
  style: {},
  containerStyle: {},
  positionRight: false,
  openAtStart: false,
};

export default Dropdown;
