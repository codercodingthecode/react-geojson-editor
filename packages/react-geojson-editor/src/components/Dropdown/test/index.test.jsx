import React from 'react';

import { shallow } from 'enzyme';
import { Dropdown } from '../component';

describe('Dropdown', () => {
  it('has not changed', () => {
    const wrapper = shallow(<Dropdown> <div className="dropdown-content"> HI </div></Dropdown>);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders', () => {
    const wrapper = shallow(<Dropdown> <div className="dropdown-content"> HI </div></Dropdown>);
    expect(wrapper.find('.toggle-button')).toHaveLength(1);
  });

  it('opens', () => {
    const wrapper = shallow(<Dropdown> <div className="dropdown-content"> HI </div></Dropdown>);
    wrapper.find('.toggle-button').simulate('click', { preventDefault: () => undefined });
    expect(wrapper.find('.dropdown-content')).toHaveLength(1);
  });

  it('closes', () => {
    const wrapper = shallow(<Dropdown> <div className="dropdown-content"> HI </div></Dropdown>);
    wrapper.find('.toggle-button').simulate('click', { preventDefault: () => undefined });
    wrapper.find('.toggle-button').simulate('click', { preventDefault: () => undefined });
    expect(wrapper.find('.dropdown-content')).toHaveLength(0);
  });
});
