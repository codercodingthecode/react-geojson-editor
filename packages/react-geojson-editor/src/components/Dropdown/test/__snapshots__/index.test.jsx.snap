// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown has not changed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    containerStyle={Object {}}
    name="Dropdown"
    style={Object {}}
  >
     
    <div
      className="dropdown-content"
    >
       HI 
    </div>
  </Dropdown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className={undefined}
        style={Object {}}
      >
        <div
          className="toggle-button"
          onClick={[Function]}
          tabIndex="0"
        >
          Dropdown
        </div>
      </div>,
      "className": undefined,
      "style": Object {},
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="toggle-button"
            onClick={[Function]}
            tabIndex="0"
          >
            Dropdown
          </div>,
          undefined,
        ],
        "className": undefined,
        "style": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dropdown",
            "className": "toggle-button",
            "onClick": [Function],
            "tabIndex": "0",
          },
          "ref": null,
          "rendered": "Dropdown",
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className={undefined}
          style={Object {}}
        >
          <div
            className="toggle-button"
            onClick={[Function]}
            tabIndex="0"
          >
            Dropdown
          </div>
        </div>,
        "className": undefined,
        "style": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="toggle-button"
              onClick={[Function]}
              tabIndex="0"
            >
              Dropdown
            </div>,
            undefined,
          ],
          "className": undefined,
          "style": Object {},
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Dropdown",
              "className": "toggle-button",
              "onClick": [Function],
              "tabIndex": "0",
            },
            "ref": null,
            "rendered": "Dropdown",
            "type": "div",
          },
          undefined,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
